models/base/scratch/bigquery/snowplow_ecommerce_base_events_this_run.sql

```diff
@@ -0,0 +1,93 @@
+{{
+  config(
+    tags=["this_run"]
+  )
+}}
+
+{%- set lower_limit, upper_limit = snowplow_utils.return_limits_from_model(ref('snowplow_ecommerce_base_sessions_this_run'),
+                                                                          'start_tstamp',
+                                                                          'end_tstamp') %}
+
+
+select
+  *,
+  row_number() over (partition by ev.domain_sessionid order by ev.derived_tstamp) AS page_view_in_session_index,
+
+from (
+  -- without downstream joins, it's safe to dedupe by picking the first event_id found.
+  select
+    array_agg(e order by e.collector_tstamp limit 1)[offset(0)].*
+
+  from (
+
+  select
+    a.contexts_com_snowplowanalytics_snowplow_web_page_1_0_0[safe_offset(0)].id as page_view_id,
+    b.domain_userid,
+
+    -- unpacking the ecommerce user object
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=user_fields(),
+        col_prefix='contexts_ecommerce_tracking_user_1',
+        relation=source('atomic', 'events'),
+        relation_alias='a') }},
+
+    -- unpacking the ecommerce checkout step object
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=checkout_step_fields(),
+        col_prefix='contexts_ecommerce_tracking_checkout_step_1',
+        relation=source('atomic', 'events'),
+        relation_alias='a') }},
+
+    -- unpacking the ecommerce page object
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=tracking_page_fields(),
+        col_prefix='contexts_ecommerce_tracking_page_1',
+        relation=source('atomic', 'events'),
+        relation_alias='a') }},
+
+    -- unpacking the ecommerce transaction object
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=transaction_fields(),
+        col_prefix='contexts_ecommerce_tracking_transaction_1',
+        relation=source('atomic', 'events'),
+        relation_alias='a') }},
+
+    -- unpacking the ecommerce cart object
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=cart_fields(),
+        col_prefix='contexts_ecommerce_tracking_cart_1',
+        relation=source('atomic', 'events'),
+        relation_alias='a') }},
+
+    -- unpacking the ecommerce action object
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=tracking_action_fields(),
+        col_prefix='unstruct_event_ecommerce_tracking_action_1',
+        relation=source('atomic', 'events'),
+        relation_alias='a') }},
+
+    a.*
```

Comment:
This gives a duplicate `domain_userid` field you'll need to add it to the exclude